src/utils/settings.rb

```diff
@@ -455,13 +467,62 @@ def write_impl_file(file)
         buf << "};\n"
 
         # Write word list
-        buf << "static const char *settingNamesWords[] = {\n"
-        buf << "\tNULL,\n"
+        buf << "static const uint8_t settingNamesWords[] = {\n"
+        word_bits = 5
+        # We can keep up to 5 extra symbols outside a-z
+        rem_symbols = 5
+        symbols = Array.new
+        acc = 0
+        acc_bits = 0
+        encode_byte = lambda do |c|
+            if c == 0
+                chr = 0 # XXX: Remove this if we go for explicit lengths
+            elsif c >= 'a'.ord && c <= 'z'.ord
+                chr = 1 + (c - 'a'.ord)
+            elsif c >= 'A'.ord && c <= 'Z'.ord
+                raise "Cannot encode uppercase character #{c.ord} (#{c})"
+            else
+                idx = symbols.index(c)
+                if idx.nil?
+                    if rem_symbols == 0
```

Comment:
Is `rem_symbols` ever decremented?

Reply:
Nice catch! That’s definitely a bug, I’ll fix it.